website/content/v1.10/kubernetes-guides/upgrading-kubernetes.md

```diff
@@ -108,6 +108,42 @@ If the command fails for any reason, it can be safely restarted to continue the
 
 > Note: When using custom/overridden Kubernetes component images, use flags `--*-image` to override the default image names.
 
+### Synchronize Declared and Deployed Configurations
+
+When Kubernetes is upgraded with `talosctl upgrade-k8s`, the live machine configuration on your nodes is updated with new component image versions (API server, controller manager, scheduler, kube-proxy, etc.).
+
+If you are storing full machine configuration files (`controlplane.yaml`, `worker.yaml`) in Git, these versions will drift out of sync.
+
+Re-applying those stale files later could unintentionally downgrade components.
+
+> **Note**: Storing full machine configurations is not recommended.
+>
+> Version numbers for Talos, etcd, Kubernetes components, and add-ons will change frequently. Maintaining these across upgrades requires editing many fields by hand.
+
+#### Recommended Workflow
+
+Instead of storing full machine configs, keep only the following:
```

Comment:
you also need to keep initial version contract, that is value of `--talos-version` flag, and never change it

also you need additional inputs: cluster name and cluster endpoint, and kubernetes version

and I think this section should go somewhere to a separate document, and just link it from here